pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -616,6 +619,36 @@ func (e *etcd) SetOwnerCheckConfig(ownerCheckConfig *OwnerCheckConfig) {
 	e.ownerCheckConfig = ownerCheckConfig
 }
 
+func (e *etcd) ScaleETCD(ctx context.Context, replicas int) error {
+	etcdObj := &druidv1alpha1.Etcd{}
+	if err := e.client.Get(ctx, client.ObjectKeyFromObject(e.etcd), etcdObj); err != nil {
+		return err
+	}
+
+	if expectedTimestamp, ok := e.etcd.Annotations[v1beta1constants.GardenerTimestamp]; ok {
+		if err := health.ObjectHasAnnotationWithValue(v1beta1constants.GardenerTimestamp, expectedTimestamp)(etcdObj); err != nil {
+			return err
+		}
+	}
+
+	if _, ok := etcdObj.Annotations[v1beta1constants.GardenerOperation]; ok {
+		return fmt.Errorf("etcd object still has operation annotation set")
+	}
+
+	patch := client.MergeFrom(etcdObj.DeepCopy())
```

Comment:
Only relevant if my first comment is not an option:

Why are you working on a new etcd object here? Components should keep working on the object they already have in memory.
IIUC, this func is always called after `Deploy`, hence `e.etcd` is already initialized with the last-known (relevant) state.

Reply:
Ok, thanks for explaining 👍